### PR TITLE
Add function forkExistingRepoR

### PR DESCRIPTION
**Pull request by erikd**
Closes: https://github.com/phadej/github/issues/280

Hacked together a little one-off test and it worked as expected.

### CODE REVIEW

File: src/GitHub/Endpoints/Repos.hs [hunk 1]
```diff
@@ -172,6 +173,12 @@ createRepoR :: NewRepo -> Request 'RW Repo
 createRepoR nrepo =
     command Post ["user", "repos"] (encode nrepo)
 
+-- | Fork an existing repository.
+-- See <https://developer.github.com/v3/repos/forks/#create-a-fork>
+forkExistingRepoR :: Name Owner -> Name Repo -> Maybe (Name Owner) -> Request 'RW Repo
+forkExistingRepoR owner repo morg =
+    command Post ["repos", toPathPart owner, toPathPart repo, "forks" ] (maybe mempty encode morg)
```
Comment (phadej) [inline src/GitHub/Endpoints/Repos.hs]: ah, the third argument seems to be *query string parameter*, if I read the documentation right.
Yet, we don't have support for that (`Command` constructor doesn't take `QueryString`).

It's up to you to decide, whether you
- add a TODO: note that third argument is unused
- alter `SimpleRequest`, I don't think there are anything complicated, compiler should guide you.

Comment (erikd) [inline src/GitHub/Endpoints/Repos.hs]: Should I add new constructor or should I modify the existing `Command`? Sorry to seem a little dense here.

Comment (phadej) [inline src/GitHub/Endpoints/Repos.hs]: Add a field to `Command`.

There's smart constructor `command`, I have no opinion whether it should be modified, or if new smart constructor is better option.

Comment (erikd) [inline src/GitHub/Endpoints/Repos.hs]: Went the "TODO" route instead. The other option seems like a bit of a rabbit hole.

**Comment by erikd** [comment]
Took the easy way out and just added a TODO.

**Pull request merged**